### ISSUE TITLE
Integration with Prometheus Operator

### ISSUE THREAD

**Issue by bcollard** [solo-io/gloo#5722]
More and more users are using **Prometheus Operator** in place of the legacy Prometheus app.

The main Custom Resources (CRs) for **Prometheus Operator** are:
![image](https://user-images.githubusercontent.com/9626829/146562683-4804f81d-9036-43f5-97f1-212949145a68.png)
See also https://prometheus-operator.dev/docs/operator/design/

While the `Prometheus` and `AlertManager` CRs should be fully managed by the platform admins, we should help with the `ServiceMonitor` CRs and also the `PrometheusRules` CRs.

The `ServiceMonitor` CRs are now provided, [starting from Gloo Edge v1.10](https://github.com/solo-io/gloo/issues/1773).
We should now implement minimalist `PrometheusRules` CRs, so that our customers can have alerting rules OOTB.

Also, we could drop the embedded versions of Prometheus and Grafana, since it's only for demo purposes today. cc. @chrisgaun

**Comment by dmpe** [comment]
Yes, please, just a new servicemonitor object in the helm chart would be more than enough.

**Comment by dmpe** [comment]
Hmm, very disappointed: https://github.com/kgateway-dev/kgateway/pull/12100
